libs/sdk-core/src/persist/swap.rs

```diff
@@ -416,24 +448,34 @@ impl SqliteStorage {
     }
 }
 
+fn append_and_clause(s: &mut String, clause: &str) {
```

Comment:
Perhaps it is better to use a slice of conditions and then when formatting the query use string join with "AND" ?

Reply:
Done